### PR TITLE
Properly report break conditions

### PR DESCRIPTION
**Pull request by jannic**
A break condition is also a framing error. As it is the more specific code, return the break error.

### CODE REVIEW

**Comment by jannic** [comment]
I wonder what error should be returned if both FE and PE are set. Is one of them more specific and should be preferred?

**Comment by jannic** [comment]
embassy-rp uses this order of precedence, which looks sensible:
```
            if dr.oe() {
                Err(nb::Error::Other(Error::Overrun))
            } else if dr.be() {
                Err(nb::Error::Other(Error::Break))
            } else if dr.pe() {
                Err(nb::Error::Other(Error::Parity))
            } else if dr.fe() {
                Err(nb::Error::Other(Error::Framing))
            } else {
                Ok(dr.data())
            }
```

**Comment by 9names** [comment]
> I wonder what error should be returned if both FE and PE are set. Is one of them more specific and should be preferred?

FWIW I would have frame error as a higher priority.
Sure, your data is corrupt either way but a parity error only tells you that an odd number of bits are wrong, a framing error tells you *why* the sum of bits is wrong.

**Comment by jannic** [comment]
I don't agree that the framing error tells you _why_ the parity was wrong. Both flags can be set because of a single bit-error. (If one, or an odd number of, data/parity bits are wrong, you get a parity error. If the stop bit is wrong, you get a framing error.)
But I agree that the framing error should probably have priority: It's more specific. You exactly know _which_ bit was wrong. So it tells you _slightly_ more than the parity error.
Now the difference between the two choices is so small that consistency with other implementations seems to be more important to me. I'll check what the C SDK does.

File: rp2040-hal/src/uart/reader.rs [hunk 1]
```diff
@@ -133,19 +133,17 @@ pub(crate) fn read_raw<'b, D: UartDevice>(
...
                 error = Some(ReadErrorType::Break);
-            }
-
-            if read.pe().bit_is_set() {
+            } else if read.pe().bit_is_set() {
                 error = Some(ReadErrorType::Parity);
-            }
-
-            if read.fe().bit_is_set() {
+            } else if read.fe().bit_is_set() {
```
Comment (ithinuel) [inline rp2040-hal/src/uart/reader.rs:146]: Is the parity bit relevant if there's a framing error?

Comment (jannic) [reply rp2040-hal/src/uart/reader.rs:146]: I didn't find error handling code in the C sdk. If both FE and PE are set, embassy reports the parity error, most (but not all) linux serial drivers I checked as well. So there doesn't seem to be a consensus on how to handle that case. And honestly, I think it doesn't matter much.

If there are random bit errors, PE will happen more often than FE. If parity is misconfigured, there will only PE errors (and each character will trigger it). If no parity bit is configured, PE is irrelevant. In general, both errors indicate that the character is probably damaged, so error handling for both is probably identical anyway.

**Review by ithinuel — APPROVED** [review]

**Pull request merged**